types/reflect.go

```diff
@@ -22,7 +22,14 @@ func deserializeType(d *Deserializer) reflect.Type {
 
 func serializeAny(s *Serializer, t reflect.Type, p unsafe.Pointer) {
 	if serde, ok := s.serdes.serdeOf(t); ok {
+		offset := len(s.b)
+		s.b = append(s.b, 0, 0, 0, 0) // store a 32-bit size placeholder
 		serde.ser(s, t, p)
+		size := len(s.b) - offset
+		if int64(size) > math.MaxUint32 {
+			panic("overflow")
```

Comment:
looking forward to the day we hit this panic, it'll be a fun use case :P

Reply:
If we have a custom serializer emitting more than 4GB for a single object we probably have bigger problems to deal with. Let's use 8 bytes though and let a compressor deal with the extra zeroes.

Reply:
Totally, it wasn't a critique, just really excited to figure out what the first use case to hit this limit will be :)